### PR TITLE
EZP-25626: Add React.js to PlatformUI app and reuse it in Alloy Editor

### PR DESCRIPTION
**Pull request by sunpietro**
https://jira.ez.no/browse/EZP-25626

Added React.js directly into main app template file: `shell.html.twig`.

### CODE REVIEW

**Comment by sunpietro** [comment]
ping @dpobel 
 I found that adding React.js by creating YUI3 module makes AlloyEditor unstable. Sometimes the editor files are loaded before React.js sometimes they are loaded loaded after.

**Comment by sunpietro** [comment]
It's related to: ezsystems/PlatformUIAssetsBundle#12

**Comment by sunpietro** [comment]
ping @dpobel

**Comment by sunpietro** [comment]
@dpobel I don't know why behat tests are failing.

File: Resources/public/js/external/ez-alloyeditor.js [hunk 1]
```diff
@@ -9,6 +9,4 @@ YUI.add('ez-alloyeditor', function (Y) {
     Y.namespace('eZ');
 
     Y.eZ.AlloyEditor = AlloyEditor;
-    Y.eZ.React = Y.eZ.AlloyEditor.React;
```
Comment (dpobel) [inline Resources/public/js/external/ez-alloyeditor.js]: removing those lines is BC break, so we need to keep the definition of `Y.eZ.React` and `Y.eZ.ReactDOM` somehow.

**Comment by dpobel** [comment]
> I found that adding React.js by creating YUI3 module makes AlloyEditor unstable. Sometimes the editor files are loaded before React.js; sometimes they are loaded loaded after it.

I guess you added a `react` module in yui.yml pointing to `bundles/ezplatformuiassets/vendors/react/react.js` and you added this module as a dependency to `alloyeditor` or similar approach. And indeed, this might not work because by default the YUI Loader loads the file asynchronously and then for files which content is not wrapped in a `YUI.add`, the code might not be executed in the expected order (in your case alloy-editor-no-react.js could be evaluated before react.js is fully loaded).

Maybe to avoid that, you could try to adopt a similar strategy as for the REST Client or AlloyEditor itself. Basically, you'll add a `react` module to yui.yml pointing to `bundles/ezplatformuiassets/vendors/react/react.js` and then you'll also add a `ez-react` module which requires `react` (and defines Y.eZ.React and Y.eZ.ReactDOM see https://github.com/ezsystems/PlatformUIBundle/pull/534/files#r57878901). Then `ez-react` should be added as a dependency of `alloyeditor`. Then if a component needs React alone, then it should depend on `ez-react`.

File: Resources/public/js/external/ez-alloyeditor.js (hunk 1 above)
Comment (andrerom) [inline Resources/public/js/external/ez-alloyeditor.js]: but we can deprecate and remove in v2 if needed

**Comment by sunpietro** [comment]
ping @dpobel

**Pull request closed without merging**